src/teleop_twist_joy.cpp

```diff
@@ -46,188 +46,156 @@ namespace teleop_twist_joy
+  std::map<std::string, std::map<std::string, double>> scale_angular_map;
 
   bool sent_disable_msg;
 };
 
 /**
  * Constructs TeleopTwistJoy.
- * \param nh NodeHandle to use for setting up the publisher and subscriber.
- * \param nh_param NodeHandle to use for searching for configuration parameters.
  */
-TeleopTwistJoy::TeleopTwistJoy(rclcpp::Node::SharedPtr & node)
+TeleopTwistJoy::TeleopTwistJoy() : Node("teleop_twist_joy_node")
 {
-  pimpl_ = new Impl();
-
-  pimpl_->node = node;
+  pimpl_ = new Impl;
 
-  pimpl_->cmd_vel_pub = node->create_publisher<geometry_msgs::msg::Twist>("cmd_vel",
-    rmw_qos_profile_sensor_data);
-  pimpl_->joy_sub = node->create_subscription<sensor_msgs::msg::Joy>("joy",
+  pimpl_->cmd_vel_pub = this->create_publisher<geometry_msgs::msg::Twist>("cmd_vel",
+    rmw_qos_profile_default);
+  pimpl_->joy_sub = this->create_subscription<sensor_msgs::msg::Joy>("joy",
     std::bind(&TeleopTwistJoy::Impl::joyCallback, this->pimpl_, std::placeholders::_1),
-    rmw_qos_profile_sensor_data);
-
-  pimpl_->enable_button = 5;
-  node->get_parameter("enable_button", pimpl_->enable_button);
-
-  pimpl_->enable_turbo_button = -1;
-  node->get_parameter("enable_turbo_button", pimpl_->enable_turbo_button);
-
-  // TODO(clalancette): node->get_parameter(s) doesn't seem to
-  // support getting a map of values yet.  Revisit this once it does.
-  // if (nh_param->getParam("axis_linear", pimpl_->axis_linear_map))
-  // {
-  //   nh_param->getParam("axis_linear", pimpl_->axis_linear_map);
-  //   nh_param->getParam("scale_linear", pimpl_->scale_linear_map);
-  //   nh_param->getParam("scale_linear_turbo", pimpl_->scale_linear_turbo_map);
-  // }
-  // else
+    rmw_qos_profile_default);
+
+  this->get_parameter_or_set("enable_button", pimpl_->enable_button, 5);
+  this->get_parameter_or_set("enable_turbo_button", pimpl_->enable_turbo_button, -1);
+  this->get_parameters("axis_linear", pimpl_->axis_linear_map);
+  if (pimpl_->axis_linear_map.count("x") == 0)
   {
     pimpl_->axis_linear_map["x"] = 5;
-    node->get_parameter("axis_linear", pimpl_->axis_linear_map["x"]);
-    pimpl_->scale_linear_map["x"] = 0.5;
-    node->get_parameter("scale_linear", pimpl_->scale_linear_map["x"]);
-    pimpl_->scale_linear_turbo_map["x"] = 1.0;
-    node->get_parameter("scale_linear_turbo", pimpl_->scale_linear_turbo_map["x"]);
+    this->set_parameter_if_not_set("axis_linear.x", pimpl_->axis_linear_map["x"]);
   }
-
-  // TODO(clalancette): node->get_parameter(s) doesn't seem to
-  // support getting a map of values yet.  Revisit this once it does.
-  // if (nh_param->getParam("axis_angular", pimpl_->axis_angular_map))
-  // {
-  //   nh_param->getParam("axis_angular", pimpl_->axis_angular_map);
-  //   nh_param->getParam("scale_angular", pimpl_->scale_angular_map);
-  //   nh_param->getParam("scale_angular_turbo", pimpl_->scale_angular_turbo_map);
-  // }
-  // else
+  this->get_parameters("axis_angular", pimpl_->axis_angular_map);
```

Comment:
The problem is the way in which these are accessed.  Just looking at `axis_angular`, for instance, the rules are generally:

1.  If `roll` or `pitch` are specified in the parameters when the node is launched, set them in the `axis_angular_map`.  Otherwise, they don't exist.
2.  If `yaw` is specified in the parameters when the node is launched, set them in `axis_angular_map`.  If unspecified, set it anyway to the default value of 5.
3.  When a joystick message arrives, lookup into `axis_angular_map` for each of `roll`, `pitch`, `yaw` to get their axis.  If it doesn't exist in the map, then just fill in 0.0 in the corresponding field in the outgoing `cmd_vel` message.

Thus, to implement what you are proposing, we'd have to change the logic for axes lookup to look for a default value (say, -1), instead of just whether it exists in the map.  While I'm not fundamentally opposed to doing that, it is a departure from upstream https://github.com/ros-teleop/teleop_twist_joy/blob/indigo-devel/src/teleop_twist_joy.cpp#L74 , so I think I'd prefer to leave this as-is for now.  We can always revisit later (both here and upstream), since it isn't an API change, just an implementation change.